wrappers/rust/icicle-cuda-runtime/src/device_context.rs

```diff
@@ -1,27 +1,39 @@
+use crate::device::set_device;
 use crate::memory::CudaMemPool;
 use crate::stream::CudaStream;
 
-/// Properties of the device used in icicle functions.
+pub const DEFAULT_DEVICE_ID: usize = 0;
+
+/// Properties of the device used in Icicle functions.
 #[repr(C)]
 #[derive(Debug, Clone)]
 pub struct DeviceContext<'a> {
-    /// Stream to use. Default value: 0.
+    /// Stream to use. Default value: 0. //TODO: multiple streams per device ?
     pub stream: &'a CudaStream, // Assuming the type is provided by a CUDA binding crate
 
     /// Index of the currently used GPU. Default value: 0.
     pub device_id: usize,
 
-    /// Mempool to use. Default value: 0.
+    /// Mempool to use. Default value: 0. //TODO: multiple mempools per device ?
     pub mempool: CudaMemPool, // Assuming the type is provided by a CUDA binding crate
 }
 
 pub fn get_default_device_context() -> DeviceContext<'static> {
+    get_default_context_for_device(DEFAULT_DEVICE_ID)
+}
+
+// TODO: CudaResult
```

Comment:
It would be nice to add the doc-comment saying this will call set_device

Reply:
trying to remove it so we only set_device explicitly